### PR TITLE
Add formatError function to graphqlExpress options

### PR DESCRIPTION
**Pull request by mlcohen**
Added a `formatError` for graphqlExpress

The format error function adds meaningful extensions to each error that are useful for clients to better diagnose and handle errors. Formatted error adhere to section 7.1.2 of the GraphQL specification https://graphql.github.io/graphql-spec/June2018/#sec-Errors

### CODE REVIEW

File: src/routes/graphql.js [hunk 1]
```diff
@@ -23,6 +23,54 @@ function injectExtensions(req) {
...
+    // Include the HTTP status code and trace ID if they exist. These can come from the underlying
+    // HTTP library such as axios. Including this information in the error for the benefit of the
+    // client that made the request.
+    //
+    // The `extensions` field conforms with the GraphQL format error specification, section 7.1.2
+    // @see https://graphql.github.io/graphql-spec/June2018/#sec-Errors
+
+    // N.B. Need to create a new error in order to avoid any potential read-only issue trying to
+    // modify the given error
+    const newError = new Error();
```
Comment (pelegdvir) [inline src/routes/graphql.js]: Let's do it theb

File: src/routes/graphql.js [hunk 2]
```diff
@@ -23,6 +23,54 @@ function injectExtensions(req) {
...
+function formatError(error) {
+    const extensions = error.extensions || {};
+    const originalError = error.originalError || {};
+    const code = extensions.code || originalError.code;
+    const headers = originalError.headers || {};
+    const traceId = (
+        extensions.traceId ||
+        originalError.traceId ||
+        headers['x-trace-id'] ||
+        headers['x-request-id']
```
Comment (pelegdvir) [inline src/routes/graphql.js]: Why do we want to mask request-id as trace-id?

File: src/routes/graphql.js (hunk 1 above)
Comment (mlcohen) [reply src/routes/graphql.js]: do it in "the b"?

File: src/routes/graphql.js (hunk 2 above)
Comment (mlcohen) [reply src/routes/graphql.js]: yeah, can split it out

File: src/routes/graphql.js (hunk 1 above)
Comment (GabeLambda) [reply src/routes/graphql.js]: I think he meant "then".

**Pull request merged**